instance/models/openedx_instance.py

```diff
@@ -415,12 +410,18 @@ def terminate_obsolete_appservers(self, days=2):
     def shut_down(self):
         """
         Shut down this instance.
-
-        This process consists of two steps:
-
-        1) Disable New Relic monitors.
-        2) Terminate all app servers belonging to this instance.
         """
-        self.disable_monitoring()
-        for appserver in self.appserver_set.all():
+        if self.load_balancing_server is not None:
+            load_balancer = self.load_balancing_server
+            self.load_balancing_server = None
+            self.save()
+            if self.active_appserver is None:
+                # If an appserver is active, reconfiguring the load_balancer happens
+                # implicitly when terminate_vm() is called further down.
+                load_balancer.reconfigure()
+        self.remove_dns_records()
+        self.deprovision_mysql()
+        self.deprovision_mongo()
+        self.deprovision_swift()
```

Comment:
@smarnach By "keeping the previous order" I was referring to the order of steps in the `delete` method, which used to terminate app servers first and then call the `deprovision_*` methods (see https://github.com/open-craft/opencraft/pull/145/commits/651acde17f536e53ca304600c3aadae795b781c4#diff-b66ef4710ed40f1b980e3ab3a34ec812L276).

Re: the question about deprovisioning databases when shutting down an instance, I think it makes sense to do that. The `shut_down` method only gets called automatically for instances associated with a merged PR, and there is a grace period of 7 days to save any data in case it is needed beyond the EOL of an instance. Also forwarding this question to @antoviaque though, to see what he thinks :)